packages/visualizations/src/components/MapPoi/Map.ts

```diff
@@ -2,46 +2,155 @@ import type { BBox } from 'geojson';
     }
 
+    /** Execute queued functions */
     private enqueue(map: maplibregl.Map) {
         this.queuedFunctions.forEach((fn) => fn(map));
         this.queuedFunctions = [];
     }
 
+    /**
+     * Event handler for click events on the map.
+     * Currently, is only used to handle popup display.
+     * @param {MapLayerMouseEvent} event
+     */
+    private onClick({ point }: MapLayerMouseEvent) {
+        this.queue((map) => {
+            /**
+             * Get features closed to the click.
+             * We ask for features that are not in base style layers
+             */
+            const features = map.queryRenderedFeatures(point, { layers: this.layerIds });
+
+            // Will close the popup
+            if (this.popupFeatures.length) return;
+
+            // If features from selected layers, update the popup
+            if (features.length) {
+                this.popupFeatures = features;
+                this.setPopup(map);
+            }
+        });
+    }
+
+    private bindedOnClick = this.onClick.bind(this);
+
+    /** Event handler for popup close event. */
+    private onPopupClose() {
+        this.popupFeatures.forEach(({ source, sourceLayer, id }) => {
+            this.queue((map) => {
+                map.setFeatureState({ source, sourceLayer, id }, { 'popup-feature': false });
+            });
+        });
+        this.popupFeatures = [];
+    }
+
+    private bindedOnPopupClose = this.onPopupClose.bind(this);
+
+    /** Set the popup content and positioning */
+    private setPopup(map: maplibregl.Map) {
+        if (!this.popupFeatures.length) return;
+
+        // Current rule: use the first feature to build the popup.
+        // TO DO: Create a menu to display a list of feature to choose from.
+        const {
+            id: featureId,
+            layer: { id: layerId },
+            geometry,
+            properties,
+            source,
+            sourceLayer,
+        } = this.popupFeatures[0];
+
+        if (geometry.type !== 'Point') return;
+
+        // Get the popup configuration for a layer
+        const popupConfiguration = this.popupsConfiguration[layerId];
+
+        // If no popup configuration for a layer, we remove the popup
+        if (!popupConfiguration) {
+            this.popup.remove();
+            this.popupFeatures = [];
+            return;
+        }
+
+        const { display, getContent, getLoadingContent } = popupConfiguration;
+
+        if (this.popup.isOpen() === false) {
+            this.popup.setLngLat(geometry.coordinates.slice() as LngLatLike).addTo(map);
+        }
+
+        this.popup.setHTML(getLoadingContent());
+
+        getContent(featureId, properties).then((content) => {
+            this.popup.setHTML(content);
+        });
+
+        const classnameModifier = display === 'sidebar' ? 'addClassName' : 'removeClassName';
+        this.popup[classnameModifier](`${POPUP_OPTIONS.className}--as-sidebar`);
```

Comment:
Wouldn't it be easier to create our own display in this case? And decide in svelte wether we want to display our own or generate a maplibre popup. This would make a useful component I think (and scope css a bit more).

But this works well and is quite clean, so if you feel like it's not worth the hassle, I can understand.

Reply:
Yes, I think we can all agree that a `MapLibre.Popup` to manage the behaviour of the sidebar is not the best option. 

At the moment, there's no clear direction in which the design wants to go with this sidebar.

IMO, we could keep it as it is. It's a simple implementation and can easily be removed later.